pyspark/export_to_parquet.py

```diff
@@ -0,0 +1,190 @@
+    " --static-partitions will be ignored.",
+)
+parser.add_argument(
+    "--select",
+    default=["*"],
+    dest="select",
+    nargs="+",
+    help="A list of all fields to include in the output. Passed to df.select(*_) before"
+    " --drop is processed and after --where is processed.",
+)
+parser.add_argument(
+    "--static-partitions",
+    dest="static_partitions",
+    default=[],
+    nargs="+",
+    help="Static partitions specified as FIELD=VALUE that will be appended to"
+    " --destination after table, with FIELD dropped from the input.",
+)
+parser.add_argument(
+    "--submission-date",
+    dest="submission_date",
+    help="Short for: --filter \"submission_date = DATE 'SUBMISSION_DATE'\""
+    ' --static-partitions submission_date=SUBMISSION_DATE --where "submission_date IS'
+    ' NOT NULL"',
+)
+parser.add_argument(
+    "--where",
+    dest="where",
+    default="TRUE",
+    help="An expression to limit the output. Passed to df.where(_). If a field is"
+    " referenced in filter and later dropped via --drop or --static-partitions then it"
+    " must be referenced here.",
+)
+parser.add_argument(
+    "--write-mode",
+    dest="write_mode",
+    default="overwrite",
+    help='Passed to df.write.mode(_). Defaults to "overwrite".',
+)
+
+args = parser.parse_args()
+
+# handle --submission-date
+if args.submission_date is not None:
+    # --filter "submission_date = DATE 'SUBMISSION_DATE'"
+    args.filter.append("submission_date = DATE '" + args.submission_date + "'")
+    # --static-partitions submission_date=SUBMISSION_DATE
+    args.static_partitions.append("submission_date=" + args.submission_date)
+    # --where "submission_date IS NOT NULL"
+    args.where += " AND submission_date IS NOT NULL"
+
+# append table and --static-partitions to destination
+args.destination = "/".join(
+    [
+        re.sub("^s3://", "s3a://", args.destination).rstrip("/"),
+        re.sub("_(v[0-9]+)$", r"/\1", args.table.rsplit(".", 1).pop()),
+    ]
+    + args.static_partitions
+)
+
+# convert --static-partitions to a dict
+args.static_partitions = dict(p.split("=", 1) for p in args.static_partitions)
+
+# remove --static-partitions fields from --partition-by
+args.partition_by = [f for f in args.partition_by if f not in args.static_partitions]
+
+# add --static-partitions fields to --drop
+args.drop += args.static_partitions.keys()
+
+# convert --filter to a single string
+args.filter = " AND ".join(args.filter)
+
+if args.dry_run:
+    print(
+        """
+(
+    SparkSession.builder.appName('export_to_parquet')
+    .getOrCreate()
+    .read.format('bigquery')
+    .option('dataset', %s)
+    .option('table', %s)
+    .option('filter', %s)
+    .load()
+    .where(%s)
+    .select(*%s)
+    .drop(*%s)
+    .write.mode(%s)
+    .partitionBy(*%s)
+    .parquet(%s)
+)
+        """.strip()
+        % tuple(
+            map(
+                lambda x: x.__repr__(),
```

Comment:
The use of map and lambda here is pretty clear to me, so I don't have strong feelings, but I would have expected this to be handled using `.format` like:

```
print("""
...
    .option('dataset', {!r})
    .option('table', {!r})
    .option('filter', {!r})
...
""".format(
                    args.dataset,
                    args.table,
                    args.filter,
                    ...
  )
)
```

Reply:
i did not know about `!r`, and that's excellent, will use

Reply:
I ended up using `dedent(""" ... {dataset!r} ... """).strip().format(**vars(args))`